pymatgen/analysis/adsorption.py

```diff
@@ -599,11 +600,11 @@ def get_mi_vec(slab):
     return mvec / np.linalg.norm(mvec)
 
 
-def get_rot(slab):
+def get_rot(slab: Slab) -> SymmOp:
     """Gets the transformation to rotate the z axis into the miller index."""
     new_z = get_mi_vec(slab)
-    a, _b, _c = slab.lattice.matrix
-    new_x = a / np.linalg.norm(a)
```

Comment:
I would revert this too @janosh . As `a_len = slab.lattice.a` gives the length of lattice vector a, and `a, _b, _c = slab.lattice.matrix` extracted the vector a itself. This leads to multiple unit test errors.

Reply:
oops, my mistake

Reply:
No worries at all. Feel free to push some related commit like #3706 to this PR, and I would (try my best to) take care of the `mypy` error later.

The thing I noticed is that the default value of `None` is triggering quite a lot of `mypy` errors, for example:
```python
pymatgen/io/exciting/inputs.py:86: error: Item "None" of "Optional[Element]" has no attribute "text"  [union-attr]
pymatgen/io/exciting/inputs.py:89: error: Item "None" of "Optional[str]" has no attribute "split"  [union-attr]
pymatgen/io/exciting/inputs.py:98: error: Item "None" of "Optional[str]" has no attribute "split"  [union-attr]
pymatgen/io/exciting/inputs.py:104: error: Item "None" of "Optional[str]" has no attribute "split"  [union-attr]
pymatgen/io/exciting/inputs.py:123: error: Item "None" of "Optional[Element]" has no attribute "get"  [union-attr]
pymatgen/io/exciting/inputs.py:126: error: Item "None" of "Optional[Element]" has no attribute "get"  [union-attr]
pymatgen/io/exciting/inputs.py:129: error: Item "None" of "Optional[Element]" has no attribute "iter"  [union-attr]
pymatgen/io/exciting/inputs.py:131: error: Item "None" of "Union[str, None, Any]" has no attribute "split"  [union-attr]
```

This is quite similar to this: https://github.com/materialsproject/pymatgen/blob/05d3f9f5643c9585e15b297a58697432c9de4dcf/pymatgen/io/vasp/optics.py#L175-L187

But I guess applying `None` check is better than just suppressing the error messages (which would break the code if the args are truly `None`).